src/id5-api.js

```diff
@@ -42,35 +52,41 @@ ID5.init = function (options) {
       if (consent.isLocalStorageAllowed()) {
         utils.logInfo('Consent to access local storage and cookies is given');
 
-        if (!storedId || refreshNeeded) {
+        if (!storedResponse || !storedResponse.universal_uid || !storedResponse.signature || refreshNeeded) {
           const gdprApplies = (consentData && consentData.gdprApplies) ? 1 : 0;
           const gdprConsentString = (consentData && consentData.gdprApplies) ? consentData.consentString : '';
-          const url = `https://id5-sync.com/g/v1/${cfg.partnerId}.json`;
-          const data = { '1puid': ID5.userId || '',
-            'gdpr': gdprApplies,
-            'gdpr_consent': gdprConsentString,
-            'rf': encodeURIComponent(referer.referer),
-            'top': referer.reachedTop ? 1 : 0,
+          const url = `https://id5-sync.com/g/v2/${cfg.partnerId}.json?gdpr_consent=${gdprConsentString}&gdpr=${gdprApplies}`;
+          const signature = (storedResponse && storedResponse.signature) ? storedResponse.signature : '';
+          const pubId = (storedResponse && storedResponse.ID5ID) ? storedResponse.ID5ID : ''; // TODO: remove when 1puid isn't needed
+
+          const data = {
+            '1puid': pubId, // TODO: remove when 1puid isn't needed
+            'v': ID5.version || '',
             'o': 'api',
-            'v': ID5.version || '' };
-          if (cfg.debug) {
-            utils.logInfo('Fetching ID5 user ID from:', url, data);
-          }
+            'rf': referer.referer,
+            'u': referer.stack[0] || window.location.href,
+            'top': referer.reachedTop ? 1 : 0,
+            's': signature,
+            'pd': cfg.pd || {},
+            'nb': nb
```

Comment:
see previous comment, I would name it "nbPage".

Reply:
done